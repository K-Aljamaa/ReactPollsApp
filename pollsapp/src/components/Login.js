import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { setAuthUser } from '../store/reducers/authUser';

function Login() {
  const dispatch = useDispatch();
  const users = useSelector(state => state.users);
  const [selectedUser, setSelectedUser] = useState('');

  const handleLogin = () => {
    if (selectedUser) {
      dispatch(setAuthUser(selectedUser));
    }
  };

  return (
    <div className="min-h-screen bg-gray-100 flex items-center justify-center">
      <div className="bg-white p-8 rounded-lg shadow-md w-96">
        <h2 className="text-2xl font-bold mb-6 text-center text-gray-800">
          Employee Polls
        </h2>
        <div className="mb-4">
          <label 
            className="block text-gray-700 text-sm font-bold mb-2" 
            htmlFor="user"
          >
            Select User
          </label>
          <select
            id="user"
            className="w-full p-2 border rounded-md focus:outline-none focus:ring-2 focus:ring-blue-500"
            value={selectedUser}
            onChange={(e) => setSelectedUser(e.target.value)}
          >
            <option value="">Select a User</option>
            {Object.values(users).map(user => (
              <option key={user.id} value={user.id}>
                {user.name}
              </option>
            ))}
          </select>
        </div>
        <button
          onClick={handleLogin}
          disabled={!selectedUser}
          className={`w-full py-2 px-4 rounded-md text-white font-medium ${
            selectedUser 
              ? 'bg-blue-500 hover:bg-blue-600' 
              : 'bg-gray-300 cursor-not-allowed'
          }`}
        >
          Login
        </button>
      </div>
    </div>
  );
}

export default Login;