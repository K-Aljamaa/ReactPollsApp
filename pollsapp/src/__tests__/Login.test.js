// src/__tests__/Login.test.js
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { thunk } from 'redux-thunk'; // Note the change here
import Login from '../components/Login';

// Create mock store with middleware
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Login Component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      users: {
        sarahedo: {
          id: 'sarahedo',
          name: 'Sarah Edo'
        }
      }
    });
  });

  test('matches snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Login />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  test('handles user selection and login', () => {
    render(
      <Provider store={store}>
        <Login />
      </Provider>
    );

    const selectElement = screen.getByRole('combobox');
    fireEvent.change(selectElement, { target: { value: 'sarahedo' } });
    
    const loginButton = screen.getByText('Login');
    fireEvent.click(loginButton);

    const actions = store.getActions();
    expect(actions).toContainEqual({
      type: 'SET_AUTH_USER',
      userId: 'sarahedo'
    });
  });
});