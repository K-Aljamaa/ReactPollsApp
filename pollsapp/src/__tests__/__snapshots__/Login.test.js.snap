// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component matches snapshot 1`] = `
<div>
  <div
    class="min-h-screen bg-gray-100 flex items-center justify-center"
  >
    <div
      class="bg-white p-8 rounded-lg shadow-md w-96"
    >
      <h2
        class="text-2xl font-bold mb-6 text-center text-gray-800"
      >
        Employee Polls
      </h2>
      <div
        class="mb-4"
      >
        <label
          class="block text-gray-700 text-sm font-bold mb-2"
          for="user"
        >
          Select User
        </label>
        <select
          class="w-full p-2 border rounded-md focus:outline-none focus:ring-2 focus:ring-blue-500"
          id="user"
        >
          <option
            value=""
          >
            Select a User
          </option>
          <option
            value="sarahedo"
          >
            Sarah Edo
          </option>
        </select>
      </div>
      <button
        class="w-full py-2 px-4 rounded-md text-white font-medium bg-gray-300 cursor-not-allowed"
        disabled=""
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
